pkg/mapper/configmap/configmap.go

```diff
@@ -115,15 +115,19 @@ func ParseMap(m map[string]string) (userMappings []config.UserMapping, roleMappi
 	rawUserMappings := make([]config.UserMapping, 0)
 	userMappings = make([]config.UserMapping, 0)
 	if userData, ok := m["mapUsers"]; ok {
-		userJson, err := utilyaml.ToJSON([]byte(userData))
+		if LenientYamlEKS {
+			//use Lenient Yaml Parser
+			err = yaml.Unmarshal([]byte(userData), &rawUserMappings)
+		} else {
+			//use Strict Yaml Parser
```

Comment:
lenient/strict is not 100% accurate, the "lenient" parser is case insensitive, but it does not konw how to deal with `username: {{SessionName}}`, it expects quotes like`username: "{{SessionName}}"`

can we have a test case for the {{SessionName}} issue, I'll try to find a link to it. But basically we are using {{ in the golang template sense, but {{ also has a meaning in the yaml sense https://yaml.org/spec/1.2.2/#flow-mappings

Reply:
thank you for the comments. there are test yamls https://github.com/kubernetes-sigs/aws-iam-authenticator/tree/master/pkg/mapper/configmap/yaml. do they cover the case you mentioned? this cr is to include https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/455 on master branch, it is only on release-0.5 branch now.

> lenient/strict is not 100% accurate, the "lenient" parser is case insensitive, but it does not konw how to deal with `username: {{SessionName}}`, it expects quotes like`username: "{{SessionName}}"`
> 
> can we have a test case for the {{SessionName}} issue, I'll try to find a link to it. But basically we are using {{ in the golang template sense, but {{ also has a meaning in the yaml sense https://yaml.org/spec/1.2.2/#flow-mappings

thank you for the comments. there are test yamls https://github.com/kubernetes-sigs/aws-iam-authenticator/tree/master/pkg/mapper/configmap/yaml. do they cover the case you mentioned? this cr is to include https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/455 on master branch, it is only on release-0.5 branch now.

Reply:
yep the testes cases are here https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/497/files#diff-8b827dc0fc13d692520fe80df4d6a52e15b065812a9f4a7b906f3ded58afd865R92